vendor/bat-native-confirmations/src/bat/confirmations/internal/redeem_payment_tokens_request.cc

```diff
@@ -87,21 +86,19 @@ std::string RedeemPaymentTokensRequest::GetContentType() const {
 ///////////////////////////////////////////////////////////////////////////////
 
 base::Value RedeemPaymentTokensRequest::CreatePaymentRequestDTO(
-    const std::vector<UnblindedToken>& tokens,
-    const std::string& payload,
-    const WalletInfo& wallet_info) const {
+    const std::vector<TokenInfo>& tokens,
+    const std::string& payload) const {
   DCHECK_NE(tokens.size(), 0UL);
-  DCHECK(!wallet_info.public_key.empty());
 
   base::Value payment_credentials(base::Value::Type::LIST);
 
-  for (const auto& token : tokens) {
+  for (const auto& token_info : tokens) {
     base::Value payment_credential(base::Value::Type::DICTIONARY);
 
-    auto credential = CreateCredential(token, payload);
+    auto credential = CreateCredential(token_info, payload);
     payment_credential.SetKey("credential", base::Value(std::move(credential)));
 
-    payment_credential.SetKey("publicKey", base::Value(wallet_info.public_key));
+    payment_credential.SetKey("publicKey", base::Value(token_info.public_key));
```

Comment:
I'm trying to understand if the current behavior (before this patch) is a privacy issue; probably not because at this stage in redemption, the server has to see the wallet public key anyway so it can do the payment?

If the wallet public key were accidentally sent in earlier stages, that would be a privacy problem since it's a persistent identifier.

Reply:
@diracdeltas apologies I did not tag you earlier as got pulled onto another issue. The wallet public key was sent. Should we arrange a call with Jimmy?

Reply:
@diracdeltas: @amirsaber is going to fix the database once the fix has gone live

Reply:
No worries, I would just like to understand the scope of the bug. If the wallet public key was sent in Step 1 of https://docs.google.com/document/d/1gqe_RISTHu3VxfzpMKds__X73yO3aIgdeB5fBvp8da0/edit#heading=h.2hb43kpd1ag3, then there would be a privacy issue I think. If it was sent in Step 4, there would be no problem.

Reply:
If it was sent in Step 1, then all records of those requests which include the wallet address should be deleted.

Reply:
@diracdeltas I can confirm it was sent in step 4, thanks